### PR TITLE
Update rust to 1.0.0-alpha, cargo and rust-nightly

### PR DESCRIPTION
**Pull request by adamreid**
rust-nightly, and cargo-nightly are updated to 2015-01-20 builds

### CODE REVIEW

File: lib/autoparts/packages/rust_nightly.rb [hunk 1]
```diff
@@ -5,12 +5,12 @@ module Autoparts
   module Packages
     class RustNightly < Package
       name 'rust-nightly'
-      version '0.12.0-nightly-63fe80e1f_2014-10-08'
+      version ''
```
Comment (MaximKraev) [inline lib/autoparts/packages/rust_nightly.rb]: it must be a version defined, boxparts requires it to work properly

File: lib/autoparts/packages/rust.rb [hunk 2]
```diff
@@ -9,12 +9,12 @@ class Rust < Package
       description 'Rust: A safe, concurrent, practical language'
       category Category::PROGRAMMING_LANGUAGES
 
-      source_url 'http://static.rust-lang.org/dist/rust-0.12.0-x86_64-unknown-linux-gnu.tar.gz'
-      source_sha1 'af9011e3cdff0f4de81d8d4eefe85d518731b2b1'
+      source_url 'https://static.rust-lang.org/dist/rust-1.0.0-alpha-x86_64-unknown-linux-gnu.tar.gz'
```
Comment (MaximKraev) [inline lib/autoparts/packages/rust.rb]: is 1.0.0-alpha  the recommended latest stable ?

Comment (adamreid) [inline lib/autoparts/packages/rust.rb]: It's the first milestone where the language features are stabilized. Moving forward releases will be focused on usability improvements (library api's etc).

**Comment by MaximKraev** [comment]
Thank you, sorry for delay, everything updated and merged

**Pull request merged**